src/main/scala/com/box/castle/core/committer/states/CommittingBatch.scala

```diff
@@ -118,7 +123,17 @@ trait CommittingBatch extends CommitterActorBase
     // We commit the consumer offset here without waiting on the result, we are assuming it will succeed most of the time
     commitConsumerOffset(batch.nextOffset, metadata)
 
-    becomeFetchingData(OffsetAndMetadata(batch.nextOffset, metadata))
+    // If BatchSizeManager is enabled then goto Idling state with a specific non-zero delay
+    batchSizeManagerOption match {
+      case Some(batchSizeManager) =>
+        val delay = batchSizeManager.getDelay(committerActorId)
+        if(delay.getMillis > 0)
+          becomeIdling(OffsetAndMetadata(batch.nextOffset, metadata), delay)
+        else
+          becomeFetchingData(OffsetAndMetadata(batch.nextOffset, metadata))
```

Comment:
let's add a log line here
`log.info("$committerActorId fetching immediately because BatchSizeManager returned a delay of 0")`